File: pickerview/src/main/java/org/jaaksi/pickerview/adapter/WheelAdapter.java
package org.jaaksi.pickerview.adapter;

public interface WheelAdapter<T> {

  /**
   * Gets items count
   *
   * @return the count of wheel items
   */
  int getItemCount();

  /**
   * Gets a wheel item by index.
   *
   * @param index the item index
   * @return the wheel item text or null
   */
  T getItem(int index);
}

File: pickerview/src/main/java/org/jaaksi/pickerview/adapter/WheelAdapter.kt
package org.jaaksi.pickerview.adapter

interface WheelAdapter<T> {
    /**
     * Gets items count
     *
     * @return the count of wheel items
     */
    val itemCount: Int

    /**
     * Gets a wheel item by index.
     *
     * @param index the item index
     * @return the wheel item text or null
     */
    fun getItem(index: Int): T?
}